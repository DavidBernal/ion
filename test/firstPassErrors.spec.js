const peg = require('pegjs')
const path = require('path')
const fs = require('fs')
const glob = require('glob')
const ion = peg.buildParser(fs.readFileSync('./ion.peg', 'utf8'))
const firstPass = require('../compilers/firstPass')
const files = glob.sync(path.join(__dirname, 'fixtures/compile_errors/*.ion'))

const expectedFiles = 4

if(files.length !== expectedFiles) throw(`Expected ${expectedFiles} files`)

files.forEach((file) => {
  test(`compile error for ${path.basename(file)}`, () => {
    try{
      firstPass(fs.readFileSync(file, 'utf8'), ion.parse)
      fail()
    }catch(e){
      expect(e).toMatchSnapshot()
    }
  })
})
