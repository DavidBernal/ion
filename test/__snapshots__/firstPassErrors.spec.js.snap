// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compile error for alreadyDeclaredVarGlobal.ion 1`] = `
"[1m[31mCompile error on line 2, column 1: [39m[22m[2mVariable 'a' has already been declared globally.[22m

...
[90m 1 | [39ma = 1
[31m 2 | [39m[31m[7ma = 2[27m[39m
     ^
..."
`;

exports[`compile error for alreadyDeclaredVarWhere.ion 1`] = `
"[1m[31mCompile error on line 3, column 3: [39m[22m[2mVariable 'a' has already been declared in context started on line 1.[22m

...
[90m 1 | [39mx = a + b where
[90m 2 | [39m  a = 1,
[31m 3 |   [39m[31m[7ma = 2[27m[39m[31m,[39m
     [2m..[22m^
[90m 4 | [39m  b = 3
[90m 5 | [39m
..."
`;

exports[`compile error for multipleExports.ion 1`] = `
"[1m[31mCompile error on line 2, column 1: [39m[22m[2mMultiple exports[22m

...
[90m 1 | [39mexport 1
[31m 2 | [39m[31m[7mexport 2[27m[39m
     ^
..."
`;

exports[`compile error for wrongHeader.ion 1`] = `
"[1m[31mCompile error on line 1, column 1: [39m[22m[2mWrong header 'boo'. A correct header would be '#!/usr/bin/env ion'[22m

...
[31m 1 | [39m[31m[7m#!/usr/bin/env boo[27m[39m
     ^
..."
`;
